### PR TITLE
Fix SSLHandshakeException while closing HTTPConduit

### LINKED ISSUE #24986: SSLHandshakeException occurs while closing HTTPConduit
The following SSLHandshakeException occurs while closing HTTPConduit:
javax.net.ssl.SSLHandshakeException:
PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
at sun.security.ssl.Alert.createSSLException(Alert.java:131)
at sun.security.ssl.TransportContext.fatal(TransportContext.java:324)
..
Caused by
at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305)
at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440)
at sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:559)
at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:197)
at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1342)
at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1317)
at sun.net.www.protocol.https.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:264)
at org.apache.cxf.transport.http.URLConnectionHTTPConduit$URLConnectionWrappedOutputStream.setupWrappedStream(URLConnectionHTTPConduit.java:318)
at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.handleHeadersTrustCaching(HTTPConduit.java:1394)
at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.onFirstWrite(HTTPConduit.java:1355)
at org.apache.cxf.transport.http.URLConnectionHTTPConduit$URLConnectionWrappedOutputStream.onFirstWrite(URLConnectionHTTPConduit.java:351)
at org.apache.cxf.io.AbstractWrappedOutputStream.write(AbstractWrappedOutputStream.java:47)
at org.apache.cxf.io.AbstractThresholdOutputStream.write(AbstractThresholdOutputStream.java:69)
at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.close(HTTPConduit.java:1409)

### PR DESCRIPTION
**Pull request by sawadood**
Fixes #24986

### CODE REVIEW

**Comment by sawadood** [comment]
#build let's do a personal build!

**Comment by sawadood** [comment]
#build Let's do a personal build!

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+     * @param proxy This parameter is non-null if connection should be proxied.
+     * @param url   The target URL.
+     *
+     * @return The HttpURLConnection for the given URL.
+     * @throws IOException
+     */
+    public HttpURLConnection createConnection(TLSClientParameters tlsClientParameters,
+            Proxy proxy, URL url) throws IOException {
+
+        LOG.fine("Create HttpURLConnection to:  " + url);
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java]:
```suggestion
        LOG.fine("Create HttpURLConnection to:  " + url); // Liberty Change 
```

Can you append the `// Liberty Change` comment to all the `LOG.fine(*)` lines you've added?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+            socketFactory = tlsClientParameters.getSSLSocketFactory();
+        } else if (socketFactory == null) {
+            SSLContext ctx = null;
+            LOG.fine("SSL socketFactory is NULL");
+            if (tlsClientParameters.getSslContext() != null) {
+                // Use the SSLContext which was set
+                LOG.fine("Get SSLContext from tlsClientParameters");
+                ctx = tlsClientParameters.getSslContext();
+            } else {
+	        if (tlsClientParameters.getTrustManagers() != null &&
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java:150]:
```suggestion
                // Liberty Change Start 
	        if (tlsClientParameters.getTrustManagers() != null && 
```

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+            } else {
+	        if (tlsClientParameters.getTrustManagers() != null && 
+			tlsClientParameters.getKeyManagers() != null) {
+                   // Get sslcontext with tlsClientParameters's Trust Managers, Key Managers, etc
+                   LOG.fine("Get SSLContext with tlsClientParameters's Trust Managers, Key Managers");
+                   ctx = org.apache.cxf.transport.https.SSLUtils.getSSLContext(tlsClientParameters);
+	        }
+		else {
+                   LOG.fine("No trustManagers set on tlsClientParameters, so use Liberty's DefaultSSLSocketFactory");
+                   socketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java:158]:
```suggestion
                   socketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
                   // Liberty Change End
```

**Review by neuwerk — CHANGES_REQUESTED** [review]

**Comment by sawadood** [comment]
#build Starting personal build after code review updates

**Review by neuwerk — APPROVED** [review]
LGTM!

**Comment by sawadood** [comment]
#build trying PB again

**Comment by neuwerk** [comment]
Failures in the PB are definitely caused by this change, we will need to reevaluate how to manage this. Probably need to emulate what the JAX-RS team is doing in the SSLManager code to fix this without causing this regression.

**Pull request closed without merging**